tests/unit/test_db.py

```diff
@@ -477,23 +476,8 @@ def test_update_endpoints_with_relation(harness):
         # Set some required data before update_endpoints is called.
         for rel in [rel_id, another_rel_id]:
             user = f"relation-{rel}"
-            harness.update_relation_data(
-                rel,
-                harness.charm.app.name,
-                {
-                    "user": user,
-                    "password": password,
-                    "database": DATABASE,
-                },
-            )
-            harness.update_relation_data(
-                peer_rel_id,
-                harness.charm.app.name,
-                {
-                    user: password,
-                    f"{user}-database": DATABASE,
-                },
-            )
+            harness.charm.set_secret("app", user, password)
```

Comment:
i re-wrote this test so it betters reflect the actual way this variables are set inside the charm (using `set_secret` instead of `update_relation_data`) [here](https://github.com/canonical/postgresql-operator/blob/b4dcc6af9379b98f40cd604a9928d765ee81175f/src/relations/db.py#L198-L201)